packages/nauth0/src/server/routes/login.ts

```diff
@@ -2,10 +2,17 @@ import { NAuth0ApiRoute } from './route';
 import { createClient, createState } from '../oidc';
 import { setCookie } from 'nookies';
 import { stateCookie } from '../cookies';
+import { RedirectKey } from '../config';
 
 export const loginRoute: NAuth0ApiRoute = async (req, res, opts) => {
   const client = await createClient(opts);
-  const state = createState();
+  const state = createState(
+    req.query[RedirectKey]
+      ? {
+          [RedirectKey]: req.query[RedirectKey],
+        }
+      : {}
```

Comment:
```suggestion
{
  [redirectKey]: redirectTo
}
```

Where `redirectTo` is the result of some function call `redirectToFromReq` or something similar which is also responsible for validating (is it a string, is it a valid url) the value. If no query parameter exists it _should_ be fine to have an object:
```ts
{
  redirectTo: undefined
}
```
and the logic within callback will just fallback to the alt options. No need to worry about if the `redirectTo` query param exists (also this is better if we have more queryParameters in the future)

Reply:
I've made this change. I haven't done much validation on the redirect though (next gives us `string | string[]` already). 

It seems that this is something the developer should do?